pkg/wallet/wallet.go

```diff
@@ -215,3 +231,16 @@ func (w *BaseWallet) FindVerKey(candidateKeys []string) (int, error) {
 	}
 	return -1, cryptoutil.ErrKeyNotFound
 }
+
+// persist marshals value and saves it in store for given key
+func persist(store storage.Store, key string, value interface{}) error {
```

Comment:
this confirms my previous comment.. persist is a generic function.. it should be part of the store not the wallet

Reply:
maybe we should create a set of utility functions in the store that can be used anywhere:
Persist()
MarshalAndPersist()

The arguments are string for the key and type `interface{}` for the value so these are generic enough to be called by any store consumer. It can be done in a separate PR..